pandas/io/excel/_openpyxl.py

```diff
@@ -530,6 +539,20 @@ def _write_cells(
                             for k, v in style_kwargs.items():
                                 setattr(xcell, k, v)
 
+        if notes is None or notes_col is None:
+            return
+
+        for row_idx, (_, val) in enumerate(notes.iterrows()):
+            for col_idx, note in enumerate(val):
+                xcell = wks.cell(
+                    # first row has columns and openpyxl starts counting at 1, not 0
+                    row=row_idx + 2,
+                    column=col_idx + notes_col,  # n columns with indexes
+                )
+                if note:
```

Comment:
Is `note` here guaranteed to be a string?

Reply:
indeed it is not, that was casted in the _xlsxwriter engine but was missing here